compiler/src/dotty/tools/dotc/core/Flags.scala

```diff
@@ -20,11 +20,7 @@ object Flags {
     opaque type Flag <: FlagSet = Long
     private[Flags] def Flag(bits: Long): Flag = bits
   }
-  type FlagSet = opaques.FlagSet
-  def FlagSet(bits: Long): FlagSet = opaques.FlagSet(bits)
-  // DOTTY TODO: replace previous 2 lines with
-  //   export opaques.FlagSet
-  // once 0.17 is released and #6721 is in the bootstrap
+  export opaques.FlagSet
 
   type Flag = opaques.Flag
```

Comment:
Can this also be exported?

Reply:
`export opaques.Flag` would also export the `def.Flag` rendering it public. This is not what we want.

Reply:
Is that method not defined as `private`? What happens when you export a private method?

Reply:
Even though it is `private[Flags]` and is exported into Flags object, this is technically fine with the contract of the `private[Flags]`(namely that it should not be exposed outside `Flags`). If we don't want it in `Flags` either, shouldn't it be `private[opaques]`?

Reply:
(except that we're using it in `Flags`, e.g. in `newFlags`:

```scala
opaques.Flag(KINDFLAGS | bits), opaques.Flag(TERMS | bits), opaques.Flag(TYPES | bits)
```

This would have been simplified if we exported the method)

Reply:
An exported member is always public.

Reply:
That does create problems when doing `export x._`. It can well become an antipattern as you must now be concerned with the private details of the implementation of `x` to do the `export` correctly. I don't know of any other case where this insight into `private` members would be required.

Reply:
An export only exports members that are visible at the point of export. But then the forwarders become public. I think that's the right tradeoff.

Reply:
Yes, that makes sense